infra/src/main/kotlin/fi/fta/geoviite/infra/geocoding/GeocodingDao.kt

```diff
@@ -80,59 +81,111 @@ class GeocodingDao(
     }
 
     fun getLayoutGeocodingContextCacheKey(
+        branch: LayoutBranch,
         trackNumberId: IntId<TrackLayoutTrackNumber>,
         moment: Instant,
     ): GeocodingContextCacheKey? {
         //language=SQL
         val sql = """
             with 
-              tn as (
-                select id, version, state, deleted 
+              tn_main as (
+                select id, version, deleted, design_id
                 from layout.track_number_version
                 where id = :tn_id
                   and draft = false
+                  and design_id is null
                   and change_time <= :moment
                 order by version desc
                 limit 1
               ),
-              rl as (
-                select id, version, deleted
+              tn_design as (
```

Comment:
Menee pitkäksi näin tehtynä mutta en heti keksinyt siistimpää tapaa. Ongelma on että yhdestä row-versiosta ei voi helposti nähdä voiko se olla tarkastelulle oleellinen vai ei, vaan tietystä branchistä pitää ottaa ajanhetkelle uusin versio ja sen jälkeen päättää oliko koko branch oleellinen vai ei (tuota puitu enemmän kommenteissa tuolla Arin alkuperäisessä reviewissä). Jos joku keksii tähän siistimmän toteutuksen niin hihkaiskaa. Aika toisteiselta tämä tuntuu...